src/metabase/email/messages.clj

```diff
@@ -246,6 +250,62 @@
      :message-type :html
      :message      message-body)))
 
+(defn send-persistent-model-error-email!
+  "Format and send an email informing the user about errors in the persistent "
+  [database-id persisted-infos trigger]
+  {:pre [(seq persisted-infos)]}
+  (let [database (:database (first persisted-infos))
+        monitoring (perms/application-perms-path :monitoring)
+        db-details (perms/feature-perms-path :details :yes database-id)
+        user-ids (->> {:select   [:pgm.user_id]
+                       :from     [[:permissions_group_membership :pgm]]
+                       :join     [[:permissions_group :pg] [:= :pgm.group_id :pg.id]]
+                       :where    [:and
+                                  [:exists {:select [1]
+                                            :from [[:permissions :p]]
+                                            :where [:and
+                                                    [:= :p.group_id :pg.id]
+                                                    [:= :p.object monitoring]]}]
+
+                                  [:exists {:select [1]
+                                            :from [[:permissions :p]]
+                                            :where [:and
+                                                    [:= :p.group_id :pg.id]
+                                                    [:= :p.object db-details]]}]]
+                       :group-by [:pgm.user_id]}
+                      db/query
+                      (mapv :user_id))
+        emails (db/select-field :email User {:where [:and
+                                                     [:= :is_active true]
+                                                     (cond-> [:or
+                                                              [:= :is_superuser true]]
+                                                       (seq user-ids) (conj [:in :id user-ids]))]})
```

Comment:
This is an edge case, but it's possible for an EE instance to downgrade back to OSS. When this happens, users with DB and/or monitoring perms lose these permissions in the app, since it's an EE-only feature. But we don't run any migrations on the permissions table when downgrading happens, so they'll still show up here.

To fix this you'll need to call `metabase.public-settings.premium-features/enable-advanced-permissions?` to check whether the instance is EE and advanced perms are enabled before including these users in this query.

Reply:
Also, might be nice to put these queries in a helper function (or augment the existing `all-admin-recipients` function to allow specifying specific feature-level permissions as well).

Reply:
Added the checks and tests. Decided against the generic helper function but it's easy enough to write if a use case for a different parameter set comes up.